From b4f092f17d93a9343c917e3a20b906b693965bda Mon Sep 17 00:00:00 2001
From: Fernando Lemos <fernandotcl@gmail.com>
Date: Tue, 15 May 2012 10:28:37 -0400
Subject: [PATCH] statusicon: use absolute delta to find the nearest icon size

Otherwise, if we only look at (pixel_size - stock_size), we miss the
case when the pixel size is smaller than any provided stock size.

https://bugzilla.gnome.org/show_bug.cgi?id=673749

statusicon: Fix implicit declaration warning

gtkstatusicon.c:1333:4: warning: implicit declaration of function 'abs' [-Wimplicit-function-declaration]

Added in commit c8378099600e4279cb5c608b546bbe778c451233
---
 gtk/gtkstatusicon.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gtk/gtkstatusicon.c b/gtk/gtkstatusicon.c
index 8bcb638..5c8a591 100644
--- a/gtk/gtkstatusicon.c
+++ b/gtk/gtkstatusicon.c
@@ -28,6 +28,7 @@
 #include "config.h"
 
 #include <string.h>
+#include <stdlib.h>
 
 #include "gtkstatusicon.h"
 
@@ -1328,10 +1329,9 @@ round_pixel_size (GtkWidget *widget,
 
   for (s = GTK_ICON_SIZE_MENU; s <= GTK_ICON_SIZE_DIALOG; s++)
     {
-      if (gtk_icon_size_lookup_for_settings (settings, s, &w, &h) &&
-	  w <= pixel_size && h <= pixel_size)
+      if (gtk_icon_size_lookup_for_settings (settings, s, &w, &h))
 	{
-	  d = MAX (pixel_size - w, pixel_size - h);
+	  d = MAX (abs (pixel_size - w), abs (pixel_size - h));
 	  if (d < dist)
 	    {
 	      dist = d;
-- 
1.7.10

